buildSrc/src/main/kotlin/androidx/build/ftl/RunTestOnFTLTask.kt

```diff
@@ -153,7 +170,16 @@ abstract class RunTestOnFTLTask @Inject constructor(
                         .firstOrNull()
                         ?.outputFile
                 )
+                task.finalizedBy(copyToDistTask)
```

Comment:
Can we avoid using finalizedBy? It breaks incrementality.

Instead we should just use copyToDistTask as the anchor task and name these as such or make RunTestOnFTLTask put the results into dist directory directly without an intermediate task.

Reply:
i was actually not sure how to do this so was going to ask it.
Specifically, I noticed that we were doing `doLast` for regular tasks, so wasn't sure if there is a reason for that.
https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:buildSrc/src/main/kotlin/androidx/build/AndroidXPlugin.kt;l=501?q=getDistributionDirectory&ss=androidx%2Fplatform%2Fframeworks%2Fsupport&start=11

I'll make the change here.

Reply:
i think separate task is better since DIST is not part of build dir, idk how well it will play with caching. 
I might make the anchor task a copy task though, that will be configured by each of these so we'll not get extra tasks while still keeping the firebase run isolated. Let me try that

Reply:
oh i remembered why I used `finalizedBy`. I want it to run even when the test fails (which fails the task). If I make it a regular dependency, failure test results will not be copied to the DIST.

Reply:
Let's keep it as finalizedBy for now.